### DISCUSSION TITLE
Transaction Receipt Status as "0x0"

### DISCUSSION THREAD

**Discussion by ghost** [ConsenSys/quorum#1511] [Q&A] [answered]
**Setup Details**:
We are running GoQuorum with the high availability setup over AKS.

**Quorum Version**: 22.4.0
**Number of Parties**: 2 (say Party A and Party B)
**Quorum**: 3 pods for both parties
**Tessera**: 1 node with 2 replicas (3 pods for both parties)
**Validator**: 4 pods with Party A, none with Party B
**Other details**: IBFT consensus protocol

**Scenario**:
*Party B is executing a private transaction, with the privacy flag as 3. The transaction also emits few events. The transaction should be visible to both parties, i.e. Party A and Party B. In `privateFor` attribute, the Tessera public keys of both parties are passed.*

**Issue**:
Blocks and Transaction details are replicated to all Quorum pods. Verified using geth console method - `eth.getTransactionReceipt(txnHash)`.
However, in one of the Party B's pod, transaction status is visible as "0x0" and logs array is empty. The issue is occurring intermittently.

**1/3 pod of Party B**
![image](https://user-images.githubusercontent.com/50927130/191569873-aa07282a-27a9-4503-bb5b-c2de944b2f6d.png)

**All other pods of Party A and Party B**
Logs array is filled with the event data and transaction status is "0x1"

**Open Questions**:
1. What could be the possible reason for one of the Quorum transaction pod having status as "0x0" and logs array as empty?
2. Internally, at what stage Transaction receipt status is set as "0x0" or "0x1"? Can you please help me understand the sequence of blocks, transaction and logs syncing?
3. When transaction status is "0x1", does that guarantee transactions are written to that Quorum node/pod? 
4. At what stage, logs details are available in the transaction receipt?

**Comment by baptiste-b-pegasys** [comment]
Hello, do you have some deployment files you can provide?
About Tessera's replicas, do they share same keys?
thank you

**Reply by ghost** [reply]
Hi @baptiste-b-pegasys  

About Tessera's replicas, do they share the same keys? - *Yes, Tessera replicas share the same keys*

Deployment files:

Attached.
[deployment-files.txt](https://github.com/ConsenSys/quorum/files/9643565/deployment-files.txt)

**Reply by ghost** [reply]
@baptiste-b-pegasys : can u please suggest on the issue.

**Reply by baptiste-b-pegasys** [reply]
Hello, I will focus now on this. I want to simplify such deployment on local docker compose or/and kubernetes.

**Reply by baptiste-b-pegasys** [reply]
I've done the goquorum HA level, not yet Tessera.
My three connected nodes to the same tessera have all the logs because there are part of this member.
If you are not part of the member, you should not see the logs of a private transaction, and it is the tessera node that tells you if you have the encrypted payload or not.

In order to make the Tessera HA, I will configure it with a common database instance with the sames keys.

https://consensys.net/docs/goquorum/en/latest/configure-and-manage/configure/high-availability/?h=high

**Reply by ghost** [reply]
@baptiste-b-pegasys: Sure! 

On the above-mentioned points, we have a similar setup.

1. *My three connected nodes to the same tessera have all the logs because there are part of this member.
If you are not part of the member, you should not see the logs of a private transaction, and it is the tessera node that tells you if you have the encrypted payload or not.* - **Receipt issue is appearing for the same member, in one of the pods. The node is a member of the private transaction.**
2. *In order to make the Tessera HA, I will configure it with a common database instance with the sames keys.* - **We are also using same MySQL database instance**


**New observation:**
After deleting the chaindata from the affected Quorum pod and resyncing it, the issue gets resolved. But eventually, after a few days the issue reappears in the same pod or some other Quorum pod for some of the contracts.
Is this due to data corruption? What could be a probable reason for this? Do you have any suggestions?

**Reply by baptiste-b-pegasys** [reply]
What is your block period and current (or targeted) TPS?
I am checking the PSV behavior.

**Reply by ghost** [reply]
> What is your block period and current (or targeted) TPS?

**Block period is set as 5 sec.** 
*We post the transaction only after we get the transaction hash of the previously posted transaction. We also wait for the transaction receipt using `web3Quorum.priv.waitForTransactionReceipt` method for that transaction hash.*

**Reply by baptiste-b-pegasys** [reply]
My hypothesis is some concurrency effect between the validated and published block and the availability of the encrypted payload on tessera.

The validators are not connected to Tessera, right?

Yet, with a block period at 5 sec and not overloaded TPS, it might be something else. I am trying a similar environment and check the workflow.

**Reply by bipeen-kumar** [reply]
@baptiste-b-pegasys Yes, the validators are **not** connected to the tessera. Below is the env variable set for the validator-
``` 
 - name: PRIVATE_CONFIG
   value: ignore
 ```
 Also, this issue is affecting all our environments. So, if you cannot replicate this issue with your setup locally, we would very much appreciate having a call set up to look at our environment.

**Reply by bipeen-kumar** [reply]
@baptiste-b-pegasys Were you able to replicate the issue at your end or do you need any more information from our end?

**Reply by bipeen-kumar** [reply]
@baptiste-b-pegasys I was going through the logs and could not get anything out of it. Could you point me in some direction on what or where I should be looking into the logs or Shall I share the log files with you, will it help?

**Comment by baptiste-b-pegasys** [comment]
I've reread the process, as the hash of the private payload is stored in the public transaction when all Tessera exchanges are made, there is no concurrent issue possible, and no problem for the validators to be outside of the org too.

So I think we should take a deeper look in Tessera setup and its data access, because when GoQuorum plays the block, and is running the tx, it asks Tessera for the encrypted payload, and the only explanation that the tx has no logs is that the encrypted payload is not returned by Tessera.

I have to setup external DB for tessera in order to reproduce this. Can you remind me what kind of DB you are using?

If you have the time of the Tx and the logs of Tessera used by the node that has an issue, I will take a look.

In terms of ressources, is the database a managed service? Maybe the logs of the db would be interesting too.

**Reply by bipeen-kumar** [reply]
Thanks, @baptiste-b-pegasys  for finding this out and pointing in the direction where the actual issue might be.

1. We use managed **Azure Database for MySQL** instance as the external database
2. We use the connection configuration file for the communication between tessera and quorum.
    `PRIVATE_CONFIG=path/to/connection-config-file.toml`
    And these are the configuration parameters-
````
httpUrl = "http://tessera-loadbalancer-endpoint"
timeout = 10
httpWriteBufferSize = 0
httpReadBufferSize = 0
````
3. We are running 3 replicas of tessera behind a load balancer service.

I will try getting the logs from the database, tessera and quorum transaction node, when we get the issue.

**Reply by bipeen-kumar** [reply]
Thanks, @baptiste-b-pegasys  for finding this out and pointing in the direction where the actual issue might be.

1. We use managed **Azure Database for MySQL** instance as the external database
2. We use the connection configuration file for the communication between tessera and quorum.
    `PRIVATE_CONFIG=path/to/connection-config-file.toml`
    And these are the configuration parameters-
````
httpUrl = "http://tessera-loadbalancer-endpoint"
timeout = 10
httpWriteBufferSize = 0
httpReadBufferSize = 0
````
3. We are running 3 replicas of tessera behind a load balancer service.

I will try getting the logs from the database, tessera and quorum transaction node, when we get the issue.

**Reply by baptiste-b-pegasys** [reply]
Can you try the develop version of Tessera, there is a fix that might resolve the issue.
[DockerHub](https://hub.docker.com/layers/quorumengineering/tessera/develop/images/sha256-d2309bdeea177fd35d518fad4c7fe6e41f445ddfb90ee7000749478ee1709d31?context=explore)
PR: https://github.com/ConsenSys/tessera/pull/1489
That's linked to https://github.com/ConsenSys/quorum/discussions/1483#discussioncomment-3731017

**Reply by bipeen-kumar** [reply]
Sure, We will upgrade the tessera with this version but to test this issue we will have to wait for a week or so because as per our observation it takes around 5-7 days for the issue to resurface once the pod restarts. 

I will keep you posted on the update.

Also, has this issue been reported by other teams/people?

**Reply by baptiste-b-pegasys** [reply]
I asked about tessera data access problem, and the issue 1483 appears to be similar because it involves also a data access problem; and no it is not common.

**Reply by smvdn** [reply]
I am seeing the Quorum logs "Merkle Root check failed actual=3ae593..xxxx expect=e50ce0..xxxx" while Transaction status is "0x0" (at least on some of the transaction-nodes even though couldn't verify the logs with all). Would this cause Transaction Failure ?

**Reply by baptiste-b-pegasys** [reply]
The log comes from https://github.com/ConsenSys/quorum/blob/master/core/state_transition_pmh.go#L72

There are some checks included with Tessera so the result and impacted contracts should be the same on every member nodes.

If the merkle tree hash differs, it could mean that the state was already corrupted or that the transaction doesn't have the good payload or the metadata is corrupted. Not sure which one happens here.

There is no error, just a failed check and an early exit (`failed = true will make sure receipt is marked as "failure"`)
I am not sure it's helping to have a consistent state across the nodes.

I think it would be worth to check what the content of `type privateMessageHandler struct {` is, to check if the encrypted payload is there or not. (tessera not responding correctly hypothesis)
https://github.com/ConsenSys/quorum/blob/master/core/state_transition_pmh.go#L26

Transaction Receipt with status 0 is a failure indeed, due by this check certainly:
```
// Quorum - Privacy Enhancements
// perform privacy enhancements checks
if pmh.mustVerify() {
var exitEarly bool
exitEarly, err = pmh.verify(vmerr)
if exitEarly {
	return &ExecutionResult{
		UsedGas:    0,
		Err:        ErrPrivateContractInteractionVerificationFailed,
		ReturnData: nil,
	}, err
}
}
// End Quorum - Privacy Enhancements
```

**Reply by bipeen-kumar** [reply]
> Can you try the develop version of Tessera, there is a fix that might resolve the issue. [DockerHub](https://hub.docker.com/layers/quorumengineering/tessera/develop/images/sha256-d2309bdeea177fd35d518fad4c7fe6e41f445ddfb90ee7000749478ee1709d31?context=explore) PR: [ConsenSys/tessera#1489](https://github.com/ConsenSys/tessera/pull/1489) That's linked to [#1483 (reply in thread)](https://github.com/ConsenSys/quorum/discussions/1483#discussioncomment-3731017)

Could you please provide the dockefile with the `Azure key vault` support for this version or publish the image with the `Azure key vault` support for the suggested develop version of the tessera

**Reply by smvdn** [reply]
> The log comes from https://github.com/ConsenSys/quorum/blob/master/core/state_transition_pmh.go#L72
> 
> There are some checks included with Tessera so the result and impacted contracts should be the same on every member nodes.
> 
> If the merkle tree hash differs, it could mean that the state was already corrupted or that the transaction doesn't have the good payload or the metadata is corrupted. Not sure which one happens here.
> 
> There is no error, just a failed check and an early exit (`failed = true will make sure receipt is marked as "failure"`) I am not sure it's helping to have a consistent state across the nodes.
> 
> I think it would be worth to check what the content of `type privateMessageHandler struct {` is, to check if the encrypted payload is there or not. (tessera not responding correctly hypothesis) https://github.com/ConsenSys/quorum/blob/master/core/state_transition_pmh.go#L26
> 
> Transaction Receipt with status 0 is a failure indeed, due by this check certainly:
> 
> ```
> // Quorum - Privacy Enhancements
> // perform privacy enhancements checks
> if pmh.mustVerify() {
> var exitEarly bool
> exitEarly, err = pmh.verify(vmerr)
> if exitEarly {
> 	return &ExecutionResult{
> 		UsedGas:    0,
> 		Err:        ErrPrivateContractInteractionVerificationFailed,
> 		ReturnData: nil,
> 	}, err
> }
> }
> // End Quorum - Privacy Enhancements
> ```

@baptiste-b-pegasys  I think the payload issue might not be available as with same payload some Transaction nodes are having good data.  I was checking on the Tessera logs for any GET /transaction API failures while the issue occurs (Tessera not responding correctly hypothesis) and Quorum logs for any Transaction errors (The metadata is corrupted hypothesis.). Not seeing a traceable error from both the logs so far. Is there a keyword or log message which we can look for when this event happens on Tessera or Quorum?  Also is there any retry available when Quorum to Tessera GET /transaction API fails?

Also, it is noted that the log data is missing even if the transaction status is 0x1. So when transaction status is "0x1", does that guarantee logs are available on that Transaction Receipt?

**Reply by baptiste-b-pegasys** [reply]
new version of tessera. https://github.com/ConsenSys/tessera/releases/tag/tessera-22.10.0

```
not member -> no payload -> failed tx -> tx status 0x0 -> no logs
same state -> same tx -> same payload -> same logs
```
so logically, if you have the payload on one or more nodes and the private tx that succeeds, you should have the same logs.
But if you miss some payloads, the state is corrupted, not sure then it will get same logs.

**Reply by smvdn** [reply]
@baptiste-b-pegasys event logs not available issue recurred in one of our environments. This time as initially one transaction node was affected. While debugging further found the below difference in Quorum Transaction Node stdout 

Affected Block: **291160**
Environment Details: We had 3 Quorum transaction nodes with a Tessera Load Balancer. Behind the Load Balancer 3 Tessera Pods are available connected to MySQL external Databases. 
**Quorum logs on affected node**

```
DEBUG[11-14|14:30:58.005] Queued delivered header or block         peer=2e5f7cb5c5333327842de9ac65bcxxxxxxxxxxce3210bcfcc3533148231xxxxxxx number=291,160 hash=e4c959..041e7d  queued=1
DEBUG[11-14|14:30:58.005] Importing propagated block               peer=2e5f7cb5c5333327842de9ac65bcxxxxxxxxxxce3210bcfcc3533148231xxxxxxx number=291,160 hash=e4c959..041e7d
TRACE[11-14|14:30:58.005] BFT: loaded voting snapshot from cache   snap.number=291,159 snap.hash=0x0327a0d2dcd47015b7f047a12c635653a7327c2d49e4400852a4987ff7xxxxx snap.epoch=30000 snap.validators="[0x354EDc1044edf4a24DA24db467B67CF07c2xxxx 0x65bc5d53114ce3210BCFcC3533148231f4B4xxxx 0xbad5CB420d4982BdC033AD1FC59FaeE96c5xxxx 0xDE0A72DE49E9c47efCD02cfAf7C50221f9exxxx]" snap.votes=[]
TRACE[11-14|14:30:58.006] BFT: loaded voting snapshot from cache   snap.number=291,159 snap.hash=0x0327a0d2dcd47015b7f047a12c635653a7327c2d49e4400852a4987ff7xxxxx snap.epoch=30000 snap.validators="[0x354EDc1044edf4a24DA24db467B67CF07c2xxxx 0x65bc5d53114ce3210BCFcC3533148231f4B4xxxx 0xbad5CB420d4982BdC033AD1FC59FaeE96c5xxxx 0xDE0A72DE49E9c47efCD02cfAf7C50221f9exxxx]" snap.votes=[]
TRACE[11-14|14:30:58.006] Propagated block                         hash=e4c959..041e7d  recipients=1 duration="925.499µs"
TRACE[11-14|14:30:58.006] Propagated block                         id=7ca93d61dcaa86ea conn=inbound         number=291,160 hash=e4c959..041e7d  td=291,161
TRACE[11-14|14:30:58.006] header data not found in ancient or level db hash=e4c959..041e7d
DEBUG[11-14|14:30:58.007] Account Extra Data root                  hash=000000..000000
DEBUG[11-14|14:30:58.007] Account Extra Data root                  hash=bf42ca..e0deaa
DEBUG[11-14|14:30:58.011] AccountExtraData root after trie commit  root=56e81f..63b421
DEBUG[11-14|14:30:58.011] AccountExtraData root after trie commit  root=bf42ca..e0deaa
DEBUG[11-14|14:30:58.011] Persisted trie from memory database      nodes=0  size=0.00B    time="2.1µs"      gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=-158834.00B
DEBUG[11-14|14:30:58.011] Persisted trie from memory database      nodes=2  size=253.00B  time="18µs"       gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B
DEBUG[11-14|14:30:58.011] Inserted new block                       number=291,160 hash=e4c959..041e7d  uncles=0 txs=1 gas=0 elapsed=4.841ms     root=9bd8f1..cf0d6d
INFO [11-14|14:30:58.011] Imported new chain segment               blocks=1 txs=1 mgas=0.000 elapsed=5.388ms     mgasps=0.000 number=291,160 hash=e4c959..041e7d  dirty=0.00B
TRACE[11-14|14:30:58.011] Announced block                          hash=e4c959..041e7d  recipients=1 duration=6.328ms

```
**Quorum logs on normal node**

```
DEBUG[11-14|14:30:58.005] Queued delivered header or block         peer=7081e68f73d2396dcfa20acxxxxxxx44edf4a24da24db467b67xxxxxxx number=291,160 hash=e4c959..041e7d  queued=1
DEBUG[11-14|14:30:58.005] Importing propagated block               peer=7081e68f73d2396dcfa20acxxxxxxx44edf4a24da24db467b67xxxxxxx number=291,160 hash=e4c959..041e7d
TRACE[11-14|14:30:58.005] BFT: loaded voting snapshot from cache   snap.number=291,159 snap.hash=0x0327a0d2dcd47015b7f047a12c635653a7327c2d49e4400852a4987ff7xxxxx snap.epoch=30000 snap.validators="[0x354EDc1044edf4a24DA24db467B67CF07c2xxxx 0x65bc5d53114ce3210BCFcC3533148231f4B4xxxx 0xbad5CB420d4982BdC033AD1FC59FaeE96c5xxxx 0xDE0A72DE49E9c47efCD02cfAf7C50221f9exxxx]" snap.votes=[]
TRACE[11-14|14:30:58.006] BFT: loaded voting snapshot from cache   snap.number=291,159 snap.hash=0x0327a0d2dcd47015b7f047a12c635653a7327c2d49e4400852a4987ff7xxxxx snap.epoch=30000 snap.validators="[0x354EDc1044edf4a24DA24db467B67CF07c2xxxx 0x65bc5d53114ce3210BCFcC3533148231f4B4xxxx 0xbad5CB420d4982BdC033AD1FC59FaeE96c5xxxx 0xDE0A72DE49E9c47efCD02cfAf7C50221f9exxxx]" snap.votes=[]
TRACE[11-14|14:30:58.006] Propagated block                         hash=e4c959..041e7d  recipients=2 duration="843.799µs"
TRACE[11-14|14:30:58.006] Propagated block                         id=3e80d3a8c65f3af0 conn=inbound         number=291,160 hash=e4c959..041e7d  td=291,161
TRACE[11-14|14:30:58.006] Propagated block                         id=7b0b0c876e730d12 conn=staticdial      number=291,160 hash=e4c959..041e7d  td=291,161
TRACE[11-14|14:30:58.006] header data not found in ancient or level db hash=e4c959..041e7d
DEBUG[11-14|14:30:58.006] Account Extra Data root                  hash=000000..000000
DEBUG[11-14|14:30:58.006] Account Extra Data root                  hash=2eb035..6f9962
TRACE[11-14|14:30:58.026] Accepted connection                      addr=100.0.0.6:21032
TRACE[11-14|14:30:58.028] Verify hashes of affected contracts      expectedHashes="map[0x14a00c99ef673ffa4b310185efd8e0f6b02e812ea96d26d177b07359fcc3464e3deb90bc5d15c99334dfa873e9aade57164396655b8ab3cxxxxxxxfff:{} 0x50a1548e575894240fe9cf1896400d5ac06a718d5163db9d8114c97fccf1997ad3af943896955e61f70bac538cb1f52ba2f4de70a046e17c29bxxxx132:{} 0xb99d406c94fd8a487de47925960bb1bf68610fdb6d7513034d8b346adcd23d8166b938c4cd9ca9776db777bf38dcbe4b47axxxxxxx4f7:{} 0xd1f8824866d014d25ecb4bc16495c886b5184c9365d46685b06d55601ab7470900706d977b2012268e3ff923305dd303f86b899d5e641750daeebxxxxxx905:{} 0xee598133fa263f19c8f4b7dfb6e94a9d414212b98bad41b3c574895af570bba13cac37fd426d9f152559df519fb55a0f4c12f69bc6342xxxxxxxc8e:{}]" numberOfAffectedAddresses=5
TRACE[11-14|14:30:58.028] Privacy metadata                         affectedAddress=0x2839136EBa63897b25B39DAAF3606CE78xxxxxx metadata="&{CreationTxHash:0xb99d406c94fd8a487de47925960bb1bf68610fdb6d7513034d8b346adcd23d8166b938c4cd9ca9776db777bf38dcbe4b47axxxxxxx4f7 PrivacyFlag:3}"
TRACE[11-14|14:30:58.028] Privacy metadata                         affectedAddress=0x5C1FD8260dDF8dDC222668E51fa04Ec0xxxxxxx metadata="&{CreationTxHash:0xee598133fa263f19c8f4b7dfb6e94a9d414212b98bad41b3c574895af570bba13cac37fd426d9f152559df519fb55a0f4c12f69bc6342xxxxxxxc8e PrivacyFlag:3}"
TRACE[11-14|14:30:58.028] Privacy metadata                         affectedAddress=0xD83Fb8a5C94D8AdC4444e2bdcCEa796F88xxxxxx metadata="&{CreationTxHash:0x14a00c99ef673ffa4b310185efd8e0f6b02e812ea96d26d177b07359fcc3464e3deb90bc5d15c99334dfa873e9aade57164396655b8ab3cxxxxxxxfff PrivacyFlag:3}"
TRACE[11-14|14:30:58.028] Privacy metadata                         affectedAddress=0x534ecf46DEBbEE285Aa1053De30cB52fA8xxxxxx metadata="&{CreationTxHash:0xd1f8824866d014d25ecb4bc16495c886b5184c9365d46685b06d55601ab7470900706d977b2012268e3ff923305dd303f86b899d5e641750daeebxxxxxx905 PrivacyFlag:3}"
TRACE[11-14|14:30:58.028] Privacy metadata                         affectedAddress=0x3BaFE812b39a20f3C76285a16Cb8954bxxxxxx metadata="&{CreationTxHash:0x50a1548e575894240fe9cf1896400d5ac06a718d5163db9d8114c97fccf1997ad3af943896955e61f70bac538cb1f52ba2f4de70a046e17c29bxxxx132 PrivacyFlag:3}"
TRACE[11-14|14:30:58.028] Verify merkle root                       merkleRoot=79a530..9bb218
DEBUG[11-14|14:30:58.029] AccountExtraData root after trie commit  root=56e81f..63b421
DEBUG[11-14|14:30:58.029] AccountExtraData root after trie commit  root=2eb035..6f9962
DEBUG[11-14|14:30:58.029] Persisted trie from memory database      nodes=46 size=8.80KiB  time="201.4µs"    gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=-182894.00B
DEBUG[11-14|14:30:58.029] Persisted trie from memory database      nodes=2  size=253.00B  time="15.8µs"     gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B
DEBUG[11-14|14:30:58.030] Inserted new block                       number=291,160 hash=e4c959..041e7d  uncles=0 txs=1 gas=0 elapsed=23.276ms    root=9bd8f1..cf0d6d
INFO [11-14|14:30:58.030] Imported new chain segment               blocks=1 txs=1 mgas=0.000 elapsed=23.955ms    mgasps=0.000 number=291,160 hash=e4c959..041e7d  dirty=0.00B
TRACE[11-14|14:30:58.030] Announced block                          hash=e4c959..041e7d  recipients=0 duration=24.768ms
```
The privacy metadata retrieval is not happening on affected node. Also checked on the Tessera Logs, could find _Request /transaction_ event is there only for 2 Transaction Nodes, possibly the affected Transaction Node haven't even requested for the Private Transaction details. 

Will it be a block metadata received for the Block **291160** by the affected Transaction Node is corrupted?

Note:  All the Transaction Nodes have the same Transaction Hash on the Block.

**Reply by baptiste-b-pegasys** [reply]
It looks like that the received block is the same: same hash, same number of transaction.

I am not sure about the AccountExtraData, which is different but might be based on the node address.

In order to have the `Verify hashes of affected contracts`, here are the conditions to met:
```
func (pmh *privateMessageHandler) mustVerify() bool {
	return pmh.hasPrivatePayload && pmh.receivedPrivacyMetadata != nil && pmh.stAPI.IsPrivacyEnhancementsEnabled()
}
```

1. it's a private message
2. it has received the private payload and its metadata from Tessera
3. privacy enhancement is enabled

Normally, if you don't have the the private payload when it's a private message, it's because the node is not a member.
Yet, Tessera could be in error, or the setup is not correct, that you don't get this data even so you think you are a member.

Have you got a change to test the new tessera?
Could be an idea to add logs in the area where GoQ loads the private payload and private metadata to see what happens.

**Reply by smvdn** [reply]
We had upgraded the Tessera Image to 22.10.0 now. Will test with the latest Image and update if the issue recurs. 

Is there any option to add a retry for Quorum to Tessera API call failures?

**Reply by baptiste-b-pegasys** [reply]
If there is an error from Tessera, we could add some retry loop backoff and add more logs.

**Comment by antonydenyer** [comment, marked as answer]
fixed with https://github.com/ConsenSys/quorum/issues/1574
